### ISSUE TITLE
[proposal] no undocumented exports

### ISSUE THREAD

**Issue by antonk52** [typescript-eslint/typescript-eslint#3205]
Hi and thank you for your work. 

Typescript's LSP has a feature where it supports additional information when looking up a type of a variable including documentation from a block comment above the variable declaration. The [official docs](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) only note the support of jsdoc.

I would like to propose a new rule `no-undocumented-exports` which should enforce documentation for exported functions/variables/classes etc. I think this plugin should be especially useful for npm packages' authors.

```typescript
export const FOO = 42 // error

// Magic number
export const BAR = 42 // error

/**
 * Magic number
 */
export const BAZ = 42 // valid
```

This rule should also be configurable. The only thing I can think of is the ability to avoid documentation for functions/variables/classes

```JSON
{
  "rules": {
    "@typescript-eslint/no-undocumented-exports": ["error", {ignore: ["function", "class"]}]
  }
}
```

I am happy to contribute this rule myself just want to make sure that it is welcomed in this project🙂

**Comment by bradzacher** [comment]
Thanks for the suggestion!

I don't think this belongs in this plugin.
It's not directly related to typescript its features, and it does not require type information.
Perhaps you might find some traction in the core ESLint repo?

**Comment by armano2** [comment]
you could try using https://github.com/gajus/eslint-plugin-jsdoc

```json
"jsdoc/require-jsdoc": ["error", { "publicOnly":true }]
```

**Comment by antonk52** [comment]
Fair point, thanks for a quick response 👍
